### DISCUSSION TITLE
rdk_doc/Robot_development/quick_start/cross_compile

### DISCUSSION THREAD

**Discussion by giscus** [D-Robotics/rdk_doc#10] [General]
# rdk_doc/Robot_development/quick_start/cross_compile

本章节介绍RDK和X86平台如何通过源码安装TogetheROS.Bot。

https://developer.d-robotics.cc/rdk_doc/Robot_development/quick_start/cross_compile

**Comment by WuChao-2024** [comment]
(1)  确认拉下docker的版本正确，区别foxy和humble。
(2)  确认从GitHub拉下的代码的分支正确，区别foxy和humble，不要拉develop。
(3)  注意，如果您只是体验TROS的功能，无需源码安装，烧录最新的系统，参考手册进行功能体验即可。如果您需要二次开发，也不需要源码安装，下载对应功能包源码参考README在板端进行编译即可。

**Comment by Fredzengfafa** [comment]
如果使用的是X5的开发板，得从github上拉取，TogetheROS文件服务器上最新的2.1.3版本暂时没有对X5版本的支持，需要改动的东西比较多。

**Reply by Fredzengfafa** [reply]
X5的开发板，现在好像只有develop分支支持，2.2.0的tag里没有兼容。。。。。

**Comment by Arata2887** [comment]
wheeltec@wheeltec:~/TogetheROS_Bot$ sudo docker run -it --entrypoint="/bin/bash" -v /home/wheeltec/TogetheROS_Bot:/home/wheeltec/TogetheROS_Bot pc_tros_ubuntu22.04:v1.0.0
[sudo] password for wheeltec:
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
exec /bin/bash: exec format error

**Reply by WuChao-2024** [reply]
你好，在交叉编译章节有清晰的描述，需要使用x86机器进行交叉编译。如果您想在板子上编译，直接在环境中编译即可。

**Comment by WangSiliang1949** [comment]
X5开发板用docker安装出现这个问题
root@ubuntu:/mnt/data/kairui.wang/test# sudo docker run -it --entrypoint="/bin/bash" -v /mnt/data/kairui.wang/test:/mnt/test 4cbdb9d61e19
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
exec /bin/bash: exec format error

**Reply by WuChao-2024** [reply]
你好，在交叉编译章节有清晰的描述，需要使用x86机器进行交叉编译。如果您想在板子上编译，直接在环境中编译即可。
